### PR TITLE
Add supports of SOCKS proxy (continuation)

### LINKED ISSUE #115: need to support socks5 proxy

### PR DESCRIPTION
**Pull request by milczarekIT**
This PR is a continuation of https://github.com/mmazi/rescu/pull/125, resolves #115.

On behalof of @mmazi  added:
- Applied: throwing an `IllegalArgumentException` when the value in config is invalid.
- Added missing documentation in `README.md`

### CODE REVIEW

File: src/main/java/si/mazi/rescu/Config.java [hunk 1]
```diff
@@ -120,6 +129,10 @@ public static String getProxyHost() {
     public static Integer getProxyPort() {
         return proxyPort;
     }
+    
+    public static Type getProxyType() {
+      return proxyType;
```
Comment (mmazi) [inline src/main/java/si/mazi/rescu/Config.java]: Please fix formatting (indent - leading spaces)

File: README.md [hunk 2]
```diff
@@ -81,8 +81,9 @@ Rescu can be configured by adding a `rescu.properties` file in your classpath.
 Supported settings with example values (copy this into your `rescu.properties` file):
 
     rescu.http.readTimeoutMillis = 5000             # Read timeout in milliseconds when performing HTTP requests. The default is 30000 (30 seconds).
-    rescu.http.readProxyHost = www.example.com      # HTTP proxy host. Both host and port must be set in order to use a proxy.
-    rescu.http.readProxyPort = 80                   # HTTP proxy port. Both host and port must be set in order to use a proxy.
+    rescu.http.readProxyHost = www.example.com      # Proxy host. Both host and port must be set in order to use a proxy.
+    rescu.http.readProxyPort = 80                   # Proxy port. Both host and port must be set in order to use a proxy.
+    rescu.http.readProxyType = SOCKS                # Proxy type. Applied only when readProxyHost and readProxyPort are set. Value must be one of java.net.Proxy.Type
```
Comment (mmazi) [inline README.md]: Please list possible values explicitly (DIRECT, HTTP, SOCKS) in the comment.

File: src/main/java/si/mazi/rescu/Config.java (hunk 1 above)
Comment (milczarekIT) [reply src/main/java/si/mazi/rescu/Config.java]: Fixed.

File: README.md (hunk 2 above)
Comment (milczarekIT) [reply README.md]: ok, done

**Comment by mmazi** [comment]
Thanks!

**Pull request merged**